vis/optimizer.py

```diff
@@ -105,82 +113,130 @@ def _get_seed_input(self, seed_input):
             seed_input = np.moveaxis(seed_input, -1, 1)
         return seed_input.astype(K.floatx())
 
-    def minimize(self, seed_input=None, max_iter=200,
+    def minimize(self, seed_inputs=None, max_iter=200,
                  input_modifiers=None, grad_modifier=None,
                  callbacks=None, verbose=True):
         """Performs gradient descent on the input image with respect to defined losses.
 
         Args:
-            seed_input: An N-dim numpy array of shape: `(samples, channels, image_dims...)` if `image_data_format=
-                channels_first` or `(samples, image_dims..., channels)` if `image_data_format=channels_last`.
+            seed_inputs: A numpy array or list of numpy array.
+                The shape of an N-dim numpy array is `(samples, channels, image_dims...)` if `image_data_format=
+                channels_first`, Or it's `(samples, image_dims..., channels)` if `image_data_format=channels_last`.
                 Seeded with random noise if set to None. (Default value = None)
             max_iter: The maximum number of gradient descent iterations. (Default value = 200)
-            input_modifiers: A list of [InputModifier](vis.input_modifiers#inputmodifier) instances specifying
+            input_modifiers: A list of InputModifier, or a list of tuple of target input tensor and InputModifier.
+                [InputModifier](vis.input_modifiers#inputmodifier) instances specifying
                 how to make `pre` and `post` changes to the optimized input during the optimization process.
                 `pre` is applied in list order while `post` is applied in reverse order. For example,
                 `input_modifiers = [f, g]` means that `pre_input = g(f(inp))` and `post_input = f(g(inp))`
             grad_modifier: gradient modifier to use. See [grad_modifiers](vis.grad_modifiers.md). If you don't
                 specify anything, gradients are unchanged. (Default value = None)
-            callbacks: A list of [OptimizerCallback](vis.callbacks#optimizercallback) instances to trigger.
+            callbacks: A list of of OptimizerCallback, or a list of tuple of target tensor and OptimizerCallback.
```

Comment:
This is making the API complex. We can simplify a lot of code if we let wrt_tensor be a single tensor (which will be the general use-case). For some special case we are making the API needlessly complex.

Reply:
You're right. I'll fix it.